gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -245,13 +311,22 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     if (position == objectSize) {
       return bytesRead > 0 ? bytesRead : -1;
     }
+
+    // read request content overlaps with cached footer data
+    if ((footerContent != null) && ((position + bytesToSkipBeforeReading)
```

Comment:
nit: Since this condition is fairly complex and `position + bytesToSkipBeforeReading` is used twice, why not to use a local variable for this sum and the whole condition? (It'd make reading this part lot easier IMHO)

Reply:
Agreed